contracts/LockedPoolzData.sol

```diff
@@ -20,7 +20,11 @@ contract LockedPoolzData is LockedControl {
                 index++;
             }
         }
-        return ids;
+        uint[] memory activeIds = new uint256[](index);
+        for(uint256 i = 0; i < index; i++){
+            activeIds[i] = ids[i];
+        } 
+        return activeIds;
```

Comment:
this need to be in other function 
suggested name:
```
 KeepNElementsInArray(array[] array, uint N)
```
call this function after if (index == ids.length) return ids